### PR TITLE
add gcd for generic spells with StartRecoveryCategory=133 and missing…

### PR DESCRIPTION
**Pull request by tst12**
… CategoryRecoveryTime and add  to possesed action bar(its need for quest)

### CODE REVIEW

File: src/game/SpellAuras.cpp [hunk 1]
```diff
@@ -7284,7 +7300,7 @@ void Aura::PeriodicTick()
 
             int32 gain = pCaster->ModifyHealth(heal);
 
-            // Health Leech effects do not generate healing aggro
+            // Health Leech effects do not generate healing aggro
             if (m_modifier.m_auraname != SPELL_AURA_PERIODIC_LEECH)
```
Comment (AbaddonL4G) [inline src/game/SpellAuras.cpp]: no change here

File: src/game/SpellAuras.cpp [hunk 2]
```diff
@@ -8223,24 +8239,24 @@ void Aura::UnregisterSingleCastAura()
...
+
+    // Dispel resistance from target SPELL_AURA_MOD_DISPEL_RESIST
+    // Only affects offensive dispels
+    if (offensive && auraTarget)
+        resistChance += auraTarget->GetTotalAuraModifier(SPELL_AURA_MOD_DISPEL_RESIST);
+
+    resistChance = resistChance < 0 ? 0 : resistChance;
+    resistChance = resistChance > 100 ? 100 : resistChance;
+
+    return 100 - resistChance;
```
Comment (AbaddonL4G) [inline src/game/SpellAuras.cpp]: no change here

File: src/game/SpellAuras.cpp [hunk 3]
```diff
@@ -6769,12 +6785,12 @@ void Aura::HandleSpiritOfRedemption(bool apply, bool Real)
...
-            m_target->InterruptNonMeleeSpells(false);
-
-        // set health and mana to maximum
-        m_target->SetHealth(m_target->GetMaxHealth());
+        // interrupt casting when entering Spirit of Redemption
+        if (m_target->IsNonMeleeSpellCasted(false))
+            m_target->InterruptNonMeleeSpells(false);
+
+        // set health and mana to maximum
+        m_target->SetHealth(m_target->GetMaxHealth());
```
Comment (AbaddonL4G) [inline src/game/SpellAuras.cpp]: no change here

File: src/game/SpellAuras.cpp (hunk 1 above)
Comment (tst12) [inline src/game/SpellAuras.cpp]: no "VISUAL", its standart TAB, line endings

**Pull request merged**